### PR TITLE
Random changes

### PR DESCRIPTION
**Pull request by wm4**
Lots of changes, including these potentially controversial ones:
- use new ffmpeg API which breaks Libav
- lowering demuxer cache default sizes
- changing --deinterlace behavior

### CODE REVIEW

File: etc/input.conf [hunk 1]
```diff
@@ -143,6 +143,7 @@
 #L cycle-values loop-file "inf" "no"    # toggle infinite looping
 #ctrl+c quit 4
 #DEL script-binding osc/visibility      # cycle OSC display
+#ctrl+h cycle-values hwdec "auto" "no"  # cycle hardware decoding
```
Comment (wiiaboo) [inline etc/input.conf]: Maybe also "auto-copy"?

File: audio/aframe.c [hunk 2]
```diff
@@ -441,12 +441,15 @@ double mp_aframe_duration(struct mp_aframe *f)
 
 // Clip the given frame to the given timestamp range. Adjusts the frame size
 // and timestamp.
+// Refuses to do change spdif frames.
```
Comment (sfan5) [inline audio/aframe.c]: <s>do</s>

File: etc/input.conf (hunk 1 above)
Comment (wm4) [reply etc/input.conf]: Would that be useful?

Comment (wiiaboo) [reply etc/input.conf]: Might help with gpu-apis/context that don't support many non-copy hwdecs while decoding high framerate hevc/vp9.

File: video/filter/vf_vavpp.c [hunk 3]
```diff
@@ -376,13 +377,24 @@ static bool initialize(struct mp_filter *vf)
...
             int num = va_query_filter_caps(vf, VAProcFilterDeinterlacing, caps,
                                            VAProcDeinterlacingCount);
             if (!num)
                 continue;
+            if (p->opts->deint_type < 0) {
+                for (int n = MP_ARRAY_SIZE(deint_algorithm) - 1; n > 0; n--) {
+                    for (int x = 0; x < num; x++) {
+                        if (caps[x].type == deint_algorithm[n]) {
+                            p->opts->deint_type = n;
+                            goto found;
```
Comment (kevmitch) [inline video/filter/vf_vavpp.c]: might be worth a log message indicating which was chosen

File: DOCS/man/vf.rst [hunk 4]
```diff
@@ -478,10 +478,14 @@ Available mpv-only filters are:
 
         no
             Don't perform deinterlacing.
+        auto
+             Select the best quality deinterlacing algorithm (default). This
+             goes by the order of the options as documented, with
+             ``motion-compensated`` being considered best quality.
```
[the remark this answers is not in the document]
Comment (wm4) [reply DOCS/man/vf.rst]: Don't know. The code looks like it should work. The filter has some trace debugging logging the used fields, you could try that if you're interested.

File: etc/input.conf (hunk 1 above)
Comment (wm4) [reply etc/input.conf]: I don't see much of a point. If direct hwdec is not supported, auto should select a -copy variant too.

File: video/filter/vf_vavpp.c (hunk 3 above)
Comment (wm4) [reply video/filter/vf_vavpp.c]: Added logging of the libva algorithm identifier. Too lazy to add stringification.

File: wscript [hunk 5]
```diff
@@ -412,9 +412,9 @@ iconv support use --disable-iconv.",
 ]
 
 ffmpeg_pkg_config_checks = [
-    'libavutil',     '>= 56.6.100',
-    'libavcodec',    '>= 58.7.100',
-    'libavformat',   '>= 58.0.102',
+    'libavutil',     '>= 56.7.100',
+    'libavcodec',    '>= 58.10.100',
+    'libavformat',   '>= 58.9.100',
```
Comment (kevmitch) [inline wscript]: Is this within the scope of an interface-changes entry? I guess we have to be prepared for this to be the beginning of the end of our libav support. Or are we really banking on libav adopting this before our next release?

**Pull request merged**